app/uk/gov/hmrc/transitmovementsrouter/connectors/PersistenceConnector.scala

```diff
@@ -108,6 +124,24 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
       execute(request, movementId)
     }
 
+  override def patchMessageStatus(movementId: MovementId, messageId: MessageId, body: MessageUpdate)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): EitherT[Future, PersistenceError, Unit] =
+    EitherT(
+      executeAndExpect(
+        updateStatusRequest(movementId, messageId)
+          .transform(_.addHttpHeaders(HeaderNames.CONTENT_TYPE -> MimeTypes.JSON))
+          .withBody(Json.toJson(body)),
+        OK
+      )
+        .map(Right(_))
+        .recover {
+          case UpstreamErrorResponse(_, NOT_FOUND, _, _) => Left(MovementNotFound(movementId))
```

Comment:
Here we will be having movementId and messageId. So please create new error type inside PersistenceError like ``` case class MessageNotFound(movementId: MovementId, messageId: MessageId)        extends PersistenceError ``` then map it with ConvertError and use that here.

Reply:
From the upstream have scenario only for MovementNotFound and thats why only added for that alone here.

Reply:
we do have flexibility to change the error here so good to change as suggested.

Reply:
As discussed, it can be mapped to either one and changing it to MessageNotFound